src/back/space/src/api/space.cpp

```diff
@@ -0,0 +1,164 @@
+#include "space.hpp"
+#include "../service/service.hpp"
+#include "../../../shared/headers.hpp"
+#include "../../../shared/errors.hpp"
+#include "../../../shared/paging.hpp"
+#include "../model/space_serialize.hpp"
+
+namespace svetit::space::handlers {
+
+Space::Space(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Space::HandleRequestJsonThrow(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body,
+	server::request::RequestContext&) const
+{
+	formats::json::ValueBuilder res;
+
+	try {
+		switch (req.GetMethod()) {
+			case server::http::HttpMethod::kGet:
+				return Get(req, res);
+			case server::http::HttpMethod::kPost:
+				return Post(req, body, res);
+			case server::http::HttpMethod::kDelete:
+				return Delete(req, res);
+			case server::http::HttpMethod::kHead:
+				return Head(req, res);
+			default:
+				throw std::runtime_error("Unsupported");
+				break;
+		}
+	} catch(const errors::Unauthorized& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kUnauthorized);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::BadRequest& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::NotFound& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kNotFound);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const errors::Conflict& e) {
+		req.SetResponseStatus(server::http::HttpStatus::kConflict);
+		res["err"] = e.what();
+		return res.ExtractValue();
+	} catch(const std::exception& e) {
+		LOG_WARNING() << "Fail to process space handle with method: "
+			<< req.GetMethodStr() << " err: " << e.what();
+		res["err"] = "Fail to process space";
+		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
+	}
+
+	return res.ExtractValue();
+}
+
+formats::json::Value Space::Get(
+	const server::http::HttpRequest& req,
+	formats::json::ValueBuilder& res) const
+{
+	if (req.HasArg("id"))
+	{
+		const auto userId = req.GetHeader(headers::kUserId);
+		if (userId.empty())
+			throw errors::Unauthorized{};
+
+		const auto id = parseUUID(req, "id");
+		res = _s.GetById(id, userId);
+	}
+	else if (req.HasArg("key"))
+	{
+		const auto userId = req.GetHeader(headers::kUserId);
+		if (userId.empty())
+			throw errors::Unauthorized{};
+
+		const auto key = req.GetArg("key");
+		if (!_s.IsValidUUID(key))
+			if (!_s.CheckKeyByRegex(key))
+				throw errors::BadRequest{"Key must be valid"};
```

Comment:
см. коммент в service
```suggestion
		if (!_s.KeyWeakCheck(key))
			throw errors::BadRequest{"Key must be valid"};
```

Reply:
Готово.